clippy_lints/src/fn_param_redef_as_mutable.rs

```diff
@@ -0,0 +1,96 @@
+use rustc_lint::{EarlyLintPass, EarlyContext};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use syntax::ast::*;
+use syntax::visit::FnKind;
+use rustc_span::Span;
+use rustc_errors::DiagnosticBuilder;
+use crate::utils::{span_lint_and_then, multispan_sugg};
+use if_chain::if_chain;
+
+declare_clippy_lint! {
+    /// **What it does:** checks if any fn parameters have been assigned to a local mutable
+    /// variable.
+    ///
+    /// **Why is this bad?** reduces the complexity of the code by removing a redundant local
+    /// mutable variable.
```

Comment:
Making this mistake **is bad** because you _add_ unnecessary complexity, not reduce it ;)

Maybe something like
```suggestion
    /// **Why is this bad?** Introduces a local variables that increases complexity unnecessarily.
```